packages/aws-rfdk/lib/core/scripts/bash/installLustreClient.sh

```diff
@@ -0,0 +1,264 @@
+    local lhs="${lhs_array[$i]}"
+    local rhs="${rhs_array[$i]}"
+    if [[ $lhs =~ ^[0-9]+$ && $rhs =~ ^[0-9]+$ && $lhs -lt $rhs ]]; then
+      return 0
+    fi
+  done
+  return 1
+}
+
+function install_on_al2() {
+  architecture=$(uname -m)
+  case $architecture in
+    x86_64)   min_version="4.14.104-95.84.amzn2.x86_64";;
+    aarch64)  min_version="4.14.181-142.260.amzn2.aarch64";;
+    *) echo "ERROR: Unrecognized Amazon Linux 2 kernel version: $kernel_version"; exit 1;;
+  esac
+
+  kernel_version=$(uname -r)
+  if is_kernel_version_lower "$kernel_version" "$min_version"; then
+    echo "ERROR: Kernel version ($kernel_version) is lower than the minimum required version: $min_version"
+    exit 1
+  fi
+
+  sudo amazon-linux-extras install -y lustre2.10
+}
+
+function install_on_al1() {
+  min_version="4.14.104-78.84.amzn1.x86_64"
+
+  kernel_version=$(uname -r)
+  if is_kernel_version_lower "$kernel_version" "$min_version"; then
+    echo "ERROR: Kernel version ($kernel_version) is lower than the minimum required version: $min_version"
+    exit 1
+  fi
+
+  sudo yum install -y lustre-client
+}
+
+function install_on_rhel() {
+  # Create map of RHEL release to kernel version shell patterns
+  # See https://access.redhat.com/articles/3078
+  declare -A rhel_kernel_versions
+  rhel_kernel_versions=(
+    [7.5]='3.10.0-862.*'
+    [7.6]='3.10.0-957.*'
+    [7.7]='3.10.0-1062.*'
+    [7.8]='3.10.0-1127.*'
+    [7.9]='3.10.0-1160.*'
+    [8.1]='4.18.0-147.*'
+    [8.2]='4.18.0-193.*'
+  )
+
+  rhel_version=$(grep ^VERSION_ID= /etc/os-release | awk -F "=" '{print $2}' | sed s/\"//g)
+  kernel_version=$(uname -r)
+
+  id=$(grep ^ID= /etc/os-release | awk -F "=" '{print $2}' | sed s/\"//g)
+  if [[ $id = "centos" ]]; then
+    # Map CentOS release to the RHEL release it was based on
+    # This information can be found by referring to the CentOS release announcements
+    # For example, see announcement for CentOS 7 (1804): https://lists.centos.org/pipermail/centos-announce/2018-May/022829.html
+    #
+    # The major and minor versions of CentOS releases match that of RHEL,
+    # so we can just extract those bits and treat them as the RHEL version
+    rhel_version=$(awk '{print $4}' /etc/centos-release | awk -F "." '{print $1"."$2}')
+  fi
+
+  case $rhel_version in
+    # RHEL 7.5 and 7.6
+    # ----------------
+    7.5|7.6)
+      case $kernel_version in
+        ${rhel_kernel_versions[7.5]}) lustre_version="2.10.5";;
+        ${rhel_kernel_versions[7.6]}) lustre_version="2.10.8";;
+        *) echo "ERROR: Unsupported kernel version $kernel_version for RHEL $rhel_version"; exit 1;;
+      esac
+      sudo yum -y install "https://downloads.whamcloud.com/public/lustre/lustre-${lustre_version}/el7/client/RPMS/x86_64/kmod-lustre-client-${lustre_version}-1.el7.x86_64.rpm"
+      sudo yum -y install "https://downloads.whamcloud.com/public/lustre/lustre-${lustre_version}/el7/client/RPMS/x86_64/lustre-client-${lustre_version}-1.el7.x86_64.rpm"
```

Comment:
The AWS docs don't have an explanation for it, but doing a bit of searching and it looks like the `kmod-lustre-client` package contains kernel-specific modules for the Lustre client. Added a comment at the top of the function with this info, since we're installing `kmod` packages in multiple places here.